Nos/Service/Relay/RelaySubscriptionManager.swift

```diff
@@ -303,3 +353,21 @@ fileprivate struct WebsocketErrorEvent {
         self.nextRetry = Date(timeIntervalSince1970: Date.now.timeIntervalSince1970 + delaySeconds.doubleValue)
     }
 }
+
+/// Represents a connection to a websocket with state tracking. Utilizes a Starscream WebSocket under the hood.
+class WebSocketConnection {
+    let socket: WebSocket
+    var state: WebSocketState 
+    
+    init(socket: WebSocket, state: WebSocketState = .disconnected) {
+        self.socket = socket
+        self.state = state
+    }
+}
+
+/// The states a WebSocketConnection can be in. These states are used by `RelaySubscriptionManager` to move each 
+/// socket to the `connected` state where we can start executing requests, taking into account time to connect, errors
+/// and authentication.
+enum WebSocketState: Equatable {
+    case disconnected, connecting, connected, errored(WebSocketErrorEvent)
```

Comment:
I don't love the name `errored` but maybe it's because I don't know WebSockets all that well. I like `error` or `failed` but maybe those don't make sense here.

Reply:
🤔 I like "error" in the past tense because it matches "disconnected" and "connected". It's saying "this socket has closed on an error in the past". Does that help? I could add some doc comments to it.

Reply:
Yeah, I thought that might be the thinking. I'd love doc comments for each case.

Reply:
Oh, you already did. Thanks!